uploadgui/src/main/scala/se/lu/nateko/cp/meta/upload/FormComponents.scala

```diff
@@ -68,18 +70,41 @@ class Select[T](elemId: String, labeller: T => String, autoselect: Boolean = fal
 class FileInput(elemId: String, cb: () => Unit){
 	private val fileInput = getElementById[html.Input](elemId).get
 	private var _hash: Try[Sha256Sum] = file.flatMap(_ => fail("hashsum computing has not started yet"))
+	private var _lastModified: Double = 0
 
 	def file: Try[dom.File] = if(fileInput.files.length > 0) Success(fileInput.files(0)) else fail("no file chosen")
-	def hash: Try[Sha256Sum] = _hash
 
-	fileInput.oninput = _ => file.foreach{f =>
+	def hash: Try[Sha256Sum] = {
+		if (hasBeenModified) {
+			fail("The file has been modified, please choose the updated version")
+		} else {
+			_hash
+		}
+	}
+
+	def hasBeenModified: Boolean =
+		file.map(_.asInstanceOf[js.Dynamic].lastModified.asInstanceOf[Double]) != Success(_lastModified)
```

Comment:
Extracting `lastModified` from a `File` object is done at least twice in the code, and as it is rather "dirty" code, it should be moved to a dedicated utility function.